shell/platform/linux/fl_engine.cc

```diff
@@ -274,10 +274,20 @@ gboolean fl_engine_start(FlEngine* self, GError** error) {
   custom_task_runners.struct_size = sizeof(FlutterCustomTaskRunners);
   custom_task_runners.platform_task_runner = &platform_task_runner;
 
+  g_autoptr(GPtrArray) command_line_args =
+      g_ptr_array_new_with_free_func(g_free);
+  if (fl_dart_project_get_enable_mirrors(self->project)) {
+    g_ptr_array_add(command_line_args, g_strdup("dart-flags"));
```

Comment:
You need a dummy argument at the beginning, since the embedder API (very unfortunately) treats the array as argv and assumes the first element is the binary name rather than a flag.